lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/SecurityTokenProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.yarn.client;
+
+import org.apache.reef.tang.annotations.DefaultImplementation;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Provides security token for setting up YARN container context.
+ */
+@DefaultImplementation(UserCredentialSecurityTokenProvider.class)
+public interface SecurityTokenProvider {
+
+  /**
+   * Returns a ByteBuffer containing security tokens.
+   * @return a ByteBuffer
+   */
+  ByteBuffer getTokens();
```

Comment:
Is `ByteBuffer` the right type for these? I'd have expected `String` or `byte[]`?